SPECS/ocaml/ocaml.spec

```diff
@@ -1,45 +1,130 @@
-%global __ocaml_requires_opts -c -f '%{buildroot}%{_bindir}/ocamlrun %{buildroot}%{_bindir}/ocamlobjinfo.byte'
-%global __ocaml_provides_opts -f '%{buildroot}%{_bindir}/ocamlrun %{buildroot}%{_bindir}/ocamlobjinfo.byte'
-%global majmin %(echo %{version} | cut -d. -f1-2)
-Summary:        OCaml compiler and programming environment
+# Don't add -Wl,-dT,<build dir>
+%undefine _package_note_flags
+
+# OCaml 5.1 broke building with LTO.  A file prims.c is generated with primitive
+# function declarations, all with "void" for their parameter list.  This does
+# not match the real definitions, leading to lots of -Wlto-type-mismatch
+# warnings.  These change the output of the tests, leading to many failed tests.
+%global _lto_cflags %{nil}
+
+# OCaml has a bytecode backend that works on anything with a C
+# compiler, and a native code backend available on a subset of
+# architectures.  A further subset of architectures support native
+# dynamic linking.
+
+%ifarch %{ocaml_native_compiler}
```

Comment:
where is %{ocaml_native_compiler} coming from ?

Reply:
It seems lots of ocaml based packages use it, we actually do have that macro defined in `macros.ocaml-srpm` which is:
```
# Architectures that support the OCaml native code compiler.
%ocaml_native_compiler  aarch64 %{arm} %{ix86} ppc ppc64 ppc64le riscv64 s390x sparc sparcv9 x86_64
```